### ISSUE TITLE
 ERROR   source-google-docs:  invalid_grant

### ISSUE THREAD

**Issue by linoxyn** [cedricdelpoux/gatsby-source-google-docs#113]
It's very possible that this is more a fault with Google's OAuth system though a while back the steps to create the `GOOGLE_DOCS_TOKEN={"access_token":"ya...J0","refresh_token":"1..mE","scope":"https://www.googleapis.com/auth/drive.metadata.readonly https://www.googleapis.com/auth/documents.readonly","token_type":"Bearer","expiry_date":1598284554759}` it seems one now has to assign a *test user* to which the access is granted to in the OAuth process. Even though the token has a fair expiry date I find that this `invalid_grant` error happens too often and new credentials need to be generated.

To generate a new token is an easy process though in other projects I only needed to create this token once. Maybe this could be happening because I am now using Gatsby v3.  It may also be that Google now demands that an *OAuth consent screen* has to be set with a *test user*. These steps were not needed before. Without these steps the consent screen fails because there will not be a set *test user*.

Any guidance would be very welcome. Thank you.

Below is the error reported in the console. 

```
 ERROR 

source-google-docs:  invalid_grant

  Error: invalid_grant

  - gaxios.ts:158 Gaxios._request
    [jhadocswebsite]/[gaxios]/src/gaxios.ts:158:15

  - task_queues.js:93 processTicksAndRejections
    internal/process/task_queues.js:93:5

  - oauth2client.js:172 OAuth2Client.refreshTokenNoCache
    [jhadocswebsite]/[google-auth-library]/build/src/auth/oauth2client.js:172:21

  - oauth2client.js:259 OAuth2Client.getRequestMetadataAsync
    [jhadocswebsite]/[google-auth-library]/build/src/auth/oauth2client.js:259:17

  - oauth2client.js:332 OAuth2Client.requestAsync
    [jhadocswebsite]/[google-auth-library]/build/src/auth/oauth2client.js:332:23

  - google-drive.js:309 fetchFiles
    [jhadocswebsite]/[gatsby-source-google-docs]/utils/google-drive.js:309:15

  - google-docs.js:28 fetchDocuments
    [jhadocswebsite]/[gatsby-source-google-docs]/utils/google-docs.js:28:31

  - source-nodes.js:33 Object.exports.sourceNodes
    [jhadocswebsite]/[gatsby-source-google-docs]/utils/source-nodes.js:33:29

  - api-runner-node.js:434 runAPI
    [jhadocswebsite]/[gatsby]/src/utils/api-runner-node.js:434:16

not finished source and transform nodes - 0.356s
not finished source-google-docs - 0.280s - fetching

 ERROR 
```

**Comment by cedricdelpoux** [comment]
Thank you for reporting this @linoxyn.

Do you have examples of other projects where there are no problem?
I don't know what changed

**Comment by linoxyn** [comment]
I could try to do a test site using Gatsby v2 and hopefully I'll get something to report back within a week or so.

I have a project that has not had this issue. I have not run it for a while though in just running this in development to see if I get this error and it did not. This error seems to be random though I can't be sure of that. I will run this a few more times in the next week to see what may happen. This project was finalized using v2 though the initial token generation was done without that step on my Google Cloud Platform where I did not need to set up a *test user* on a *OAuth consent screen* The token was completely stable without this above error. I'm not sure if it's an issue within the plugin, googleapis, or within a limitation set on the app or the *test user*. thank you :)

**Comment by linoxyn** [comment]
I'm not sure though I think I've found the issue. During a few updates that were lagging going from Gatsby v2 to v3 my project encountered a few incompatible package dependencies versions and so I found I was using `npm update --force` in order to update the project. The dotenv   version  8.6.0 installed expected  version  8.6.0  though `--force`  updated the dotenv package to version 10.0.0 I will keep an eye on this for now. @cedricdelpoux if you want to close this for now or wait, I will leave that up to you. I don't know, would it be worth the dotenv update to the latest version 10.0.0 in your plugin? Thank you for your patience and concern.
